### PR TITLE
UI: Add toggle to enable/disable metric autocomplete

### PR DESCRIPTION
**Pull request by ipstatic**
This change adds a toggle to enable or disable the metric autocomplete
functionality. By default it is enabled. This is a port of a change I
did in [Thanos][1].

[1]: https://github.com/thanos-io/thanos/pull/3381

Signed-off-by: Jarod Watkins <jarod@42lines.net>

### CODE REVIEW

**Review by prmsrswt — COMMENTED** [review]
You need to run `make assets` here as well. :)

**Comment by ipstatic** [comment]
> You need to run `make assets` here as well. :)

Hmm, I ran this and it doesn't appear to have changed anything. Any ideas?

```
λ make assets
>> writing assets
# Un-setting GOOS and GOARCH here because the generated Go code is always the same,
# but the cached object code is incompatible between architectures and OSes (which
# breaks cross-building for different combinations on CI in the same container).
cd web/ui && GO111MODULE=on GOOS= GOARCH= go generate -x -v  -mod=vendor
doc.go
go run -mod=vendor assets_generate.go
ui.go
~/go/src/github.com/ipstatic/prometheus (enable-disable-metric-findahead) λ git status
On branch enable-disable-metric-findahead
nothing to commit, working tree clean
```

**Comment by roidelapluie** [comment]
It is no longer needed.

**Comment by roidelapluie** [comment]
cc @juliusv

File: web/ui/react-app/src/pages/graph/Panel.tsx [hunk 1]
```diff
@@ -233,6 +234,7 @@ class Panel extends Component<PanelProps, PanelState> {
               onExpressionChange={this.handleExpressionChange}
               executeQuery={this.executeQuery}
               loading={this.state.loading}
+              enable={this.props.enableMetricAutocomplete}
```
Comment (juliusv) [inline web/ui/react-app/src/pages/graph/Panel.tsx:237]: If I just read `enable` in the context of `ExpressionInput`, I wouldn't know what it means, as it lacks specificity / context. Could you give it a longer name here too?

**Comment by juliusv** [comment]
:+1: besides the naming nit. Note all of this will have to change when switching to https://github.com/prometheus-community/codemirror-promql eventually.

File: web/ui/react-app/src/pages/graph/Panel.tsx (hunk 1 above)
Comment (ipstatic) [reply web/ui/react-app/src/pages/graph/Panel.tsx]: Went ahead and used the full name used elsewhere.

**Comment by ipstatic** [comment]
> 👍 besides the naming nit. Note all of this will have to change when switching to https://github.com/prometheus-community/codemirror-promql eventually.

How should I approach this? Maybe just open a feature request noting we should have a way to control metric autocomplete?

**Comment by juliusv** [comment]
@ipstatic Basically whenever I get around adding that editor to Prometheus (just working on making the React UI the default now), I already have the requirement in my mind to make autocomplete, syntax highlighting, and linting all individually optional. Like it is now if you open the settings menu in https://promlens.com/

**Comment by juliusv** [comment]
:+1: Thanks!

**Pull request merged**